docs/01-app/02-building-your-application/07-configuring/05-mdx.mdx

```diff
@@ -248,6 +248,67 @@ export default function Page() {
 
 Navigating to the `/mdx-page` route should display your rendered MDX page.
 
+<AppOnly>
+
+### Using dynamic imports
+
+You can also dynamically import MDX in your pages.
```

Comment:
It would be good to explain why you might want to import MDX dynamically.

Reply:
Yeah- in my case the markdown didn't live in the app directory, but I still wanted the markdown to be part of the Next.js import graph so I could reap all benefits - specifically images.

Reply:
> in my case the markdown didn't live in the app directory, but I still wanted the markdown to be part of the Next.js import graph so I could reap all benefits - specifically images.

I guess this is a benefit, but not necessarily because of the ability to import MDX dynamically?

Reply:
I think having the content outside of the `app/` directory will be one of the most common reasons for users to look for this functionality

Maybe other benefits could be:

1. Programmatically importing some content (eg. programmatic logic beyond a simple dynamic route segment like `[slug]/page.mdx)
2. Importing MDX fragments smaller than a full page
3. Enabling greater control over the layers in between the page and the MDX content (eg. via wrapping intermediary React components)

Reply:
I'll mix in point 1, but 2-3 is also possible without the use of a dynamic import.